Adding MadCatz FLY5 Flight Stick

john-tiberius opened the pull request:
Here we go.

john-tiberius committed:
Adding MadCatz FLY5 Flight Stick

dbankieris commented:
@john-tiberius Can I borrow one of these to double-check your decoding?

dbankieris commented on CMakeLists.txt:
```diff
@@ -3,6 +3,10 @@ project(IDF)
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/CMakeListsDIR.txt)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/CMakeListsDIR.txt)
+endif()
+
```
I'm not well-versed in cmake. What's this bit about?

dbankieris replied:
@john-tiberius Hey, you! Answer my question!

dbankieris commented on include/idf/MadCatz.hh:
```diff
@@ -0,0 +1,125 @@
+/**
+ * %MadCatz hand controller's input layout
+ *
+ * @author Derek Bankieris
```
You can put yourself here or just remove this line.

dbankieris commented on include/idf/UsbMadCatz.hh:
```diff
@@ -0,0 +1,49 @@
+/**
+ * USB MadCatz hand controller
+ *
+ * @author Derek Bankieris
```
You can put yourself here or just remove this line.

dbankieris commented on source/idf/UsbMadCatz.cpp:
```diff
@@ -0,0 +1,65 @@
+void UsbMadCatz::decode(const std::vector<unsigned char>& data) {
+    uint16_t value;
+     
+    //forwardBackwardPivot.setValue(((unsigned)data[2] & 0xF) << 6 | data[1] >> 2);
```
Remove commented out code.

dbankieris commented on source/idf/UsbMadCatz.cpp:
```diff
@@ -0,0 +1,65 @@
+    value = ((uint16_t) data[2] & 0x000f) << 6 |  ((uint16_t)data[1] & 0x00FC) >> 2;
+    forwardBackwardPivot.setValue(value);
+
+    //leftRightPivot.setValue(((unsigned)data[1] & 3) << 8 | data[0]);
```
Remove commented out code.

dbankieris commented:
The decoding looks great, nice work! The only thing I'm not sure about is `scrollUp` vs `scrollDown`. When I roll my thumb up the scroll wheel, `scrollDown` goes active. Are those accidentally swapped, or am I thinking about it wrong?

john-tiberius commented:
They could be swapped, they should work as you suggest.


From: dbankieris <notifications@github.com>
Reply-To: nasa/IDF <reply@reply.github.com>
Date: Friday, March 29, 2019 at 3:34 PM
To: nasa/IDF <IDF@noreply.github.com>
Cc: "Bullock, John T. (JSC-ER7)[CACI NSS, INC]" <john.t.bullock@nasa.gov>, Mention <mention@noreply.github.com>
Subject: Re: [nasa/IDF] Adding MadCatz FLY5 Flight Stick (#27)


The decoding looks great, nice work! The only thing I'm not sure about is scrollUp vs scrollDown. When I roll my thumb up the scroll wheel, scrollDown goes active. Are those accidentally swapped, or am I thinking about it wrong?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_nasa_IDF_pull_27-23issuecomment-2D478140535&d=DwMCaQ&c=ApwzowJNAKKw3xye91w7BE1XMRKi2LN9kiMk5Csz9Zk&r=MUjJhzmeC1az98ZfAfG65PaqutODMPqfW6_SN0dcI0M&m=XXxmMRbyTT6lnO_9SRipYkLKKZP9TYO2D7vixuqgA4Y&s=QKMHcy4V5kLFV2sJ_2A59zKBFXytm9bwNXLxdoSdvno&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AucR5faeSRWBlY3v180fWiQa8kfstdRuks5vbnicgaJpZM4b8XSi&d=DwMCaQ&c=ApwzowJNAKKw3xye91w7BE1XMRKi2LN9kiMk5Csz9Zk&r=MUjJhzmeC1az98ZfAfG65PaqutODMPqfW6_SN0dcI0M&m=XXxmMRbyTT6lnO_9SRipYkLKKZP9TYO2D7vixuqgA4Y&s=HHPCC9oJowUgis-WbfdaSeXGElNYCZj5tlYXTlL-z8Y&e=>.

dbankieris committed:
Flip scrollUp and scrollDown decoding

Remove author lines
Remove commented-out code
Add MadCatz to the Configurator

dbankieris committed:
Remove checks for optional CMake files

The idea was to allow the user to override some CMake stuff with these
optional files. We can revisit this if someone requests it, but I'd like
to keep CMakeLists.txt as lean as possible for now.
